bin/packages/build-worker.js

```diff
@@ -76,109 +76,114 @@ function getBuildPath( file, buildFolder ) {
 	return path.resolve( pkgBuildPath, relativeToSrcPath );
 }
 
-/**
- * Object of build tasks per file extension.
- *
- * @type {Object<string,Function>}
- */
-const BUILD_TASK_BY_EXTENSION = {
-	async '.scss'( file ) {
-		const outputFile = getBuildPath(
-			file.replace( '.scss', '.css' ),
-			'build-style'
+async function buildCSS( file ) {
+	const outputFile = getBuildPath(
+		file.replace( '.scss', '.css' ),
+		'build-style'
+	);
+	const outputFileRTL = getBuildPath(
+		file.replace( '.scss', '-rtl.css' ),
+		'build-style'
+	);
+
+	const [ , contents ] = await Promise.all( [
+		makeDir( path.dirname( outputFile ) ),
+		readFile( file, 'utf8' ),
+	] );
+	const builtSass = await renderSass( {
+		file,
+		includePaths: [ path.join( PACKAGES_DIR, 'base-styles' ) ],
+		data:
+			[
+				'colors',
+				'breakpoints',
+				'variables',
+				'mixins',
+				'animations',
+				'z-index',
+			]
+				// Editor styles should be excluded from the default CSS vars output.
+				.concat(
+					file.includes( 'common.scss' ) ||
+						( ! file.includes( 'block-library' ) &&
+							! file.includes( 'editor-styles.scss' ) )
+						? [ 'default-custom-properties' ]
+						: []
+				)
+				.map( ( imported ) => `@import "${ imported }";` )
+				.join( ' ' ) + contents,
+	} );
+
+	const result = await postcss(
+		require( '@wordpress/postcss-plugins-preset' )
+	).process( builtSass.css, {
+		from: 'src/app.css',
+		to: 'dest/app.css',
+	} );
+
+	const resultRTL = await postcss( [ require( 'rtlcss' )() ] ).process(
+		result.css,
+		{
+			from: 'src/app.css',
+			to: 'dest/app.css',
+		}
+	);
+
+	await Promise.all( [
+		writeFile( outputFile, result.css ),
+		writeFile( outputFileRTL, resultRTL.css ),
+	] );
+}
+
+async function buildJS( file ) {
+	for ( const [ environment, buildDir ] of Object.entries(
+		JS_ENVIRONMENTS
+	) ) {
+		const destPath = getBuildPath(
+			file.replace( /\.tsx?$/, '.js' ),
```

Comment:
Nit, but it seems like this logic could belong inside `getBuildPath`.

Reply:
The `getBuildPath` function is also used by the CSS build task. There, the replacement is `.scss` -> `.css`. I don't see an easy way how to move the rules into `getBuildPath`.